singletons.go

```diff
@@ -0,0 +1,46 @@
+package lxr
+
+import "sync"
+
+// The goal of instances is to provide a way for multiple packages to use LXR without
+// instantiating multiple bytemaps in memory or having to share references
+
+var instanceMtx sync.Mutex
+var instances map[uint64]*LXRHash
+
+func init() {
+	instances = make(map[uint64]*LXRHash)
+}
+
+// Init provides access to shared instances of LXRHash without having to instantiate multiple bytemaps.
+// Two separate calls to Init(n) will result in a reference to the same object.
+// LXRHash will be instantiated with the package defaults of:
+// 	* Seed: 0xFAFAECECFAFAECEC
+// 	* Hash Size: 256
+// 	* Passes: 5
+func Init(bitsize uint64) *LXRHash {

```

Comment:
Why is this limited to package defaults?

Reply:
short answer: because that's what we're using

these are supposed to provide a "default" singleton for pegnet projects, so the goal was to make it as easy as possible to include. the only reason the bitsize is specifiable is because travis compiles choke on 30 bits

projects who want to use different parameters can still initialize the hash the normal way, though i'm not aware of any of those projects existing. for now, i wanted to keep it as simple as i could

Reply:
seems like we could use all the parameters as the keys to the hash table, and it would work for everyone rather than just our app.  Could just use the filename as the key, as we use the filenames to kinda do this already.

Reply:
I've added a branch Singleton_2  that factors out the filename code, and uses it as a key instead of just the bitsize.  It also removes Release() (per comment below).

If we keep Release() it should take the singleton as a parameter and use the singleton to reference count.  Simple enough, but I don't think Release() and reference counting is worth the effort.

Reply:
> If we keep Release() it should take the singleton as a parameter and use the singleton to reference count. Simple enough, but I don't think Release() and reference counting is worth the effort.

that's a different kind of pointer-memory system that would make sense in C but not in a language with a garbage collector

Reply:
> I've added a branch Singleton_2 that factors out the filename code, and uses it as a key instead of just the bitsize.

thoughts on making an Init() that uses the package defaults, and an InitCustom(seed, bits, size, passes)?

Reply:
I started down that route, but then thought maybe there was a reason to break out the singleton feature?